<launch>
    <arg name="mav_name" default="iris"/>
    <arg name="fcu_url" default="udp://:14540@127.0.0.1:14557"/>
    <arg name="gcs_url" default="" />
    <arg name="tgt_system" default="1" />
    <arg name="tgt_component" default="1" />
    <arg name="command_input" default="2" />
    <arg name="gazebo_simulation" default="true" />
    <arg name="visualization" default="false"/>
    <arg name="log_output" default="screen" />
    <arg name="fcu_protocol" default="v2.0" />
    <arg name="respawn_mavros" default="false" />

    <node pkg="geometric_controller" type="geometric_controller_node" name="geometric_controller" output="screen">
            <param name="mav_name" type="string" value="$(arg mav_name)" />
            <remap from="command/bodyrate_command" to="/mavros/setpoint_raw/attitude"/>
            <param name="ctrl_mode" value="$(arg command_input)" />
            <param name="enable_sim" value="$(arg gazebo_simulation)" />
            <param name="enable_gazebo_state" value="true"/>
            <param name="max_acc" value="10.0" />
            <!-- CHANGE THIS PARAMETER TO SET THE INITIAL POSITION OF THE DRONE -->
            <param name="init_pos_x" value="0" />
            <param name="init_pos_y" value="0" />
            <param name="init_pos_z" value="0" />
            <param name="Kp_x" value="8.0" />
            <param name="Kp_y" value="8.0" />
            <param name="Kp_z" value="10.0" />
            <param name="normalizedthrust_constant" value="0.037" />
            <param name="normalizedthrust_offset" value="0" />
    </node>

    <!-- launch Gazebo, ROS, Mavros, etc -->
    <node name="sim_env_setup" pkg="controllers" type="sim_env_setup.sh"/>

    <!-- launch custom node -->
    <node name="main_control" pkg="controllers" type="main_control.py" output="screen" launch-prefix="gnome-terminal --">
        <param name="bool_sim" value="True" />
        <param name="adv_ctrl" value="True" />
    </node>

</launch>